src/queue/multilevel.rs

```diff
@@ -384,6 +401,32 @@ impl LevelManager {
             level0_chance
         };
         self.level0_chance.set(new_chance);
+
+        let cur_total_tasks_per_level: [u64; LEVEL_NUM] =
+            array::from_fn(|i| self.task_poll_duration[i].get_sample_count());
+        let cur_total_tasks = cur_total_tasks_per_level.iter().sum::<u64>();
+        let last_level0_total_tasks = self.last_exec_tasks_per_level[0].get();
+        let last_total_tasks: u64 = self.last_exec_tasks_per_level.iter().map(|c| c.get()).sum();
+
+        let level_0_tasks = (cur_total_tasks_per_level[0] - last_level0_total_tasks) as usize;
+        let total_tasks = (cur_total_tasks - last_total_tasks) as usize;
+        if total_tasks > ADJUST_LEVEL_STEAL_SIZE_THRESHOLD {
```

Comment:
Sorry, how to understand this threshold? `maybe_adjust_chance` is called every second. So, the adjustment only happens when there are more than 100 tasks added in the last second?

Reply:
When there are not many tasks, we don't need to adjust the threshold as every steal can only pop 1 or 2 tasks. I moved the update current count of each level within the if block to make it more reasonable.

Reply:
I got your idea somewhat. What if there are only heavy tasks and the task incoming rate is slow? For example, there are only 50 tasks coming per second but each may run for 1 second. They will all fall in the bottom level but not trigger the adjustment (if I understand correctly)

Reply:
Yes. So I changed the logic to do not update the accumulated value when the total task is under the threshold, thus in this example, it can changed the batch-size after two round of adjust.